Svenbjorn/whatson
# Refactoring and Enhancement of Grid Expand Arrow Logic

### Why:

The primary objective was to implement the grid expand arrow feature for event, place, tour, and hiking grids.

While introducing this feature was the main focus, it presented an opportunity to revisit the underlying codebase. Though not strictly necessary, refactoring was undertaken to make the integration of the new feature smoother. This led to streamlining the codebase, enhancing maintainability, and improving the user experience with the grid expand arrow logic. As a result, we now have an optimized fetch-and-display logic, reduced redundancy, and a consistent expand arrow functionality across the platform.

- Task: [Monday.com](https://visitfaroeislands.monday.com/boards/4732496989/views/112923023/pulses/4533934471)
- Design: [Invision](https://sendistovan.invisionapp.com/console/VFI-Tourism---desktop-ckpjdr19e0hqa010ju83s0gs8/ckpjds4ht0gc0011xa3n7rndz/play)

### Changes Made:

1. **File Renaming:** Changed the main `main.js` to `whatson_main.js` and adjusted its usage to reflect the renaming.
2. **Component Introduction:** Introduced the `expand_arrow` partial component.
3. **Code Refactoring:** Logic has been refactored to be more efficient and modular.
4. **Function Management:** Moved several functions into different files for better code organization.
5. **Documentation:** All newly introduced or refactored functions have been thoroughly documented for clarity.
6. **Feature Addition:** Added `expand_arrow` logic to various sections including places, tours, and others.
7. **Optimization:** Refactored the usage of `createCloseEventHandler` to eliminate duplicate code and streamline its application.

### How to Test:

1. **File Check:** Ensure that `whatson_main.js` is present and that there are no references to the old `main.js`.
2. **Expand Arrow Logic:** Interact with the grid and ensure that the expand arrow logic functions as expected across different sections such as `places`, `tours`, etc.
  - [x] Places
  - [x] Tours
  - [x] Hiking 

4. **Functionality:** Verify that the fetch-and-display logic shows the expected results without errors.
  - [x] Places
  - [x] Tours
  - [x] Hiking 

6. **URL Behavior:** Interact with the grids and confirm that the URL updates correctly based on the logic described in the `setupCloseEventHandlers` function.
  - [x] Places
  - [x] Tours
  - [x] Hiking 

8. **Error Handling:** Ensure that in the absence of required elements or states (like Spruce's `app` store), the appropriate error messages are logged to the console.
9. **Documentation:** Review the provided documentation for clarity and completeness.
